ext/pcntl/pcntl.c

```diff
@@ -986,7 +991,7 @@ PHP_FUNCTION(pcntl_signal)
 			php_error_docref(NULL, E_WARNING, "Error assigning signal");
 			RETURN_FALSE;
 		}
-		zend_hash_index_del(&PCNTL_G(php_signal_table), signo);
+		zend_hash_index_update(&PCNTL_G(php_signal_table), signo, handle);
```

Comment:
Although, this does make me think about the handling side in pcntl_signal_dispatch.  No reason to attempt to call the handler, if it is a long.